src/utilities/_hypre_utilities.h

```diff
@@ -211,6 +211,46 @@ typedef double                 hypre_double;
  * SPDX-License-Identifier: (Apache-2.0 OR MIT)
  ******************************************************************************/
 
+#ifndef HYPRE_BASE_HEADER
+#define HYPRE_BASE_HEADER
+
+/* Base public solver struct */
+struct hypre_Solver_struct;
```

Comment:
You should be able to remove the other occurrences of this throughout the code.  It's repeated in a number of places, I think because we didn't know where to put it before (can't remember).  Here are the files:
```
krylov/HYPRE_lobpcg.h
krylov/HYPRE_krylov.h
parcsr_ls/HYPRE_parcsr_ls.h
sstruct_ls/HYPRE_sstruct_ls.h
struct_ls/HYPRE_struct_ls.h
```

Reply:
Rob, do you agree that the public base structs should go in `HYPRE_utilities.h`?

Reply:
Yes, I think so.